misc/validate_accounting.py

```diff
@@ -309,125 +365,117 @@ def __call__(self, msg):
         self.ok = False
 
 class CheckContext:
-    def __init__(self, accounts, err):
-        # accounts from indexer
-        self.accounts = accounts
+    def __init__(self, err):
         self.err = err
         self.match = 0
         self.neq = 0
         # [(addr, "err text"), ...]
         self.mismatches = []
+        self.lock = threading.Lock()
+
+    def check(self, indexer_account, algod_account):
+        niceaddr = algod_account['address']
+        address = algosdk.encoding.decode_address(niceaddr)
+        xe = ccerr(self.err)
 
-    def check(self, address, niceaddr, microalgos, assets, acfg, appparams, applocal):
-        # check data from sql or algod vs indexer
-        i2v = self.accounts.pop(address, None)
-        if i2v is None:
-            self.neq += 1
-            self.err.write('{} not in indexer\n'.format(niceaddr))
+        microalgos = algod_account['amount-without-pending-rewards']
+        indexerAlgos = indexer_account['amount-without-pending-rewards']
+        if indexerAlgos == microalgos:
+            pass # still ok
         else:
-            xe = ccerr(self.err)
-            indexerAlgos = i2v['amount-without-pending-rewards']
-            if indexerAlgos == microalgos:
-                pass # still ok
+            emsg = 'algod v={} i2 v={} '.format(microalgos, indexerAlgos)
+            if address == reward_addr:
+                emsg += ' Rewards account'
+            elif address == fee_addr:
+                emsg += ' Fee account'
+            xe(emsg)
+        assets = algod_account.get('assets')
+        i2assets = indexer_account.get('assets')
+        if i2assets:
```

Comment:
the `if{ if/else } else` logic is hard to follow. Now that we have the deleted flag can you filter the deleted fields and just compare lengths?